server/game/cards/05_LOF/units/BaylanSkollEnigmaticMaster.ts

```diff
@@ -0,0 +1,41 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { WildcardCardType } from '../../../core/Constants';
+import { CostAdjustType } from '../../../core/cost/CostAdjuster';
+import * as EnumHelpers from '../../../core/utils/EnumHelpers';
+
+export default class BaylanSkollEnigmaticMaster extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: 'baylan-skoll#enigmatic-master-id',
+            internalName: 'baylan-skoll#enigmatic-master',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addWhenPlayedAbility({
+            title: 'Use the Force to return a non-leader unit that costs 4 or less to its owner\'s hand',
+            optional: true,
+            immediateEffect: AbilityHelper.immediateEffects.useTheForce(),
+            ifYouDo: {
+                title: 'Return a non-leader unit that costs 4 or less to its owner\'s hand',
+                targetResolver: {
+                    cardCondition: (card) => card.isNonLeaderUnit() && card.cost <= 4,
+                    immediateEffect: AbilityHelper.immediateEffects.returnToHand()
+                },
```

Comment:
I would also add a `cardTypeFilter` here, this way it will say "non-leader unit" in the prompt:
```suggestion
                targetResolver: {
                    cardTypeFilter: WildcardCardType.NonLeaderUnit,
                    cardCondition: (card) => card.isNonLeaderUnit() && card.cost <= 4,
                    immediateEffect: AbilityHelper.immediateEffects.returnToHand()
                },
```